### PR TITLE
Implemented SyncInfo

### PR DESCRIPTION
**Pull request by clstrfsck**

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import com.radixdlt.serialization.SerializerDummy;
+import com.radixdlt.serialization.SerializerId2;
+import com.radixdlt.serialization.DsonOutput.Output;
+
+/**
+ * Current state of synchronisation for sending node.
+ */
+@Immutable
+@SerializerId2("consensus.sync_info")
+public final class SyncInfo {
```
Comment (talekhinezh) [inline radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java]: Can we name this something like "BFTHighQC" as imo the main purpose of this data structure is for propagating the highQC/parent for consensus. Also can we add comment highestCommittedQC->committed needs to be an ancestor of highQC->proposed. But highestCommittedQC->proposed does not need to be an ancestor of highQC->proposed.

File: radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		@JsonProperty("highest_qc") QuorumCertificate highestQC,
+		@JsonProperty("committed_qc") QuorumCertificate highestCommittedQC
+	) {
+		return new SyncInfo(highestQC, highestCommittedQC);
+	}
+
+	private SyncInfo(QuorumCertificate highestQC, QuorumCertificate highestCommittedQC) {
+		this.highestQC = Objects.requireNonNull(highestQC);
+		// Don't include separate committedQC if it is the same as highQC
+		if (highestCommittedQC == null || highestQC.equals(highestCommittedQC)) {
```
Comment (talekhinezh) [inline radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java]: Is this optimization only for serialization size? If so, should add comment.

File: radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java (hunk 1 above)
Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java]: OK, done.  Change the name to HighQC, as the BFT prefix doesn't seem apt here.

Added comments to factory method as suggested.

File: radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java (hunk 2 above)
Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/consensus/SyncInfo.java]: OK, comment added.

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**